homeassistant/components/obihai/sensor.py

```diff
@@ -148,6 +157,10 @@ def icon(self):
 
     def update(self) -> None:
         """Update the sensor."""
+        if not self._pyobihai.check_account():
```

Comment:
What does `False` mean here? If we can't connect we should set the `available` property to `False` in the entity. If credentials are bad we should start a reauth flow.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler#reauthentication

Reply:
The `available` property will return False if `self._state is None` which is what we've done here.  I'll take a look at the re-auth feedback, though I think that should be a PR in-and-of-itself as it would require new classes and tests.

Reply:
Setting the native value to `None` means that the state is unknown. This is different than state unavailable. It would be better not to couple these things to avoid confusion when reading the code.